apptools/persistence/state_pickler.py

```diff
@@ -445,12 +475,14 @@ def _do_dict(self, value):
 
     def _do_numeric(self, value):
         idx = self._register(value)
-        if PY_VER > 2:
-            data = base64.encodebytes(gzip_string(numpy.ndarray.dumps(value)))
-        else:
-            data = base64.encodestring(gzip_string(numpy.ndarray.dumps(value)))
+        data = base64_encode(gzip_string(numpy.ndarray.dumps(value)))
         return dict(type='numeric', id=idx, data=data)
 
+    def _do_numpy_generic_type(self, value):
+        idx = self._register(value)
+        self._misc_cache.append(value)
+        data = base64_encode(pickle.dumps(value))
```

Comment:
Pickling numpy scalars is quite costly (in space). I wonder if there is a better way.